driver/src/driver/stablex_driver.rs

```diff
@@ -41,14 +41,21 @@ impl<'a> StableXDriver<'a> {
             solution
         };
 
-        let submitted = if solution.executed_sell_amounts.iter().sum::<u128>() > 0 {
-            self.contract.submit_solution(batch, orders, solution)?;
-            info!("Successfully applied solution to batch {}", batch);
-            true
+        let current_objective_value = self.contract.get_current_objective_value()?;
+        let submitted = if let Some(objective_value) = solution.objective_value {
+            if current_objective_value < objective_value {
```

Comment:
How about an objective value of 0? Would that ever get submitted?

Reply:
At the moment, the contract does not allow for trivial solution submission. Furthermore, it does not accept solutions with zero objective evaluation either.